src/ui/input/color-picker/color-picker.tsx

```diff
@@ -1,51 +1,68 @@
-import React from 'react';
+import React, { PureComponent } from 'react';
 import classNames from 'classnames';
-import { CheckboxLabel, CheckboxLabelProps } from '@teambit/base-ui.input.checkbox.label';
 import styles from './color-picker.module.scss';
 import sizes from './sizes.module.scss';
-import { HiddenColorInput } from './hidden-color-input';
+import { ColorsBox, ColorsBoxProps } from './colors-box';
 
 export type ColorPickerProps = {
-  /**
-   * The color to show in the color box, and also the color value for the Input.
-   */
-  value?: string;
-  /**
-   * Control the CSS class of the indicator.
-   */
-  indicatorClassName?: string;
   /**
    * box sizes
    */
   size?: 'm' | 'l';
-} & CheckboxLabelProps;
+  disabled?: boolean;
+  /**
+   * Control the CSS class of the colors box.
+   */
+  colorsBoxClassName?: string;
+} & ColorsBoxProps;
+
+type ColorPickerState = {
+  isHidden: boolean;
+};
+
+export const DEFAULT_COLOR = '#BABEC9';
+
+export class ColorPicker extends PureComponent<ColorPickerProps, ColorPickerState> {
+  state = {
+    isHidden: true,
+  };
 
-export const DEFAULT_COLOR = '#babec9';
+  handleOpen = () => {
+    const { disabled } = this.props;
+    if (disabled) return;
+    const { isHidden } = this.state;
+    this.setState({ isHidden: !isHidden });
+  };
 
-export function ColorPicker({
-  value = DEFAULT_COLOR,
-  onInputChanged,
-  disabled,
-  className,
-  children,
-  input = <HiddenColorInput onChange={onInputChanged} disabled={disabled} value={value} />,
-  indicatorClassName,
-  size,
-  ...rest
-}: ColorPickerProps) {
-  const indicator = (
-    <div className={classNames(styles.indicator, indicatorClassName)}>
-      {children && <div className={styles.textBox}>{children}</div>}
-      <div className={styles.colorBox} style={{ backgroundColor: value }} />
-    </div>
-  );
-  return (
-    <CheckboxLabel
-      className={classNames(styles.label, sizes.pickerSizes, className)}
-      input={input}
-      indicator={indicator}
-      {...rest}
-      data-size={size}
-    />
-  );
+  render() {
+    const {
+      value = DEFAULT_COLOR,
+      size,
+      disabled,
+      colorsBoxClassName,
+      onSelect,
+      className,
+      children,
+      ...rest
+    } = this.props;
+    const { isHidden } = this.state;
+    return (
+      <div
```

Comment:
use our existing Dropdown (it will save you css and state)

Reply:
`@teambit/base-ui.surfaces.drawer`, etc

Reply:
You right.

Reply:
I prefer to use the one from evangelist.

Reply:
done.

Reply:
👍